File: backend-dotnet/EventModelingGitHubCloneDotNet/EventModelingGitHubCloneDotNet/Slices/Read/PullRequestComments/Presentation/PullRequestCommentsController.cs
using System;
using System.Threading.Tasks;
using EventModelingGitHubCloneDotNet.Slices.Read.PullRequestComments.Core.Application;
using Microsoft.AspNetCore.Mvc;

namespace EventModelingGitHubCloneDotNet.Slices.Read.PullRequestComments.Presentation
{
    [ApiController]
    [Route("/rest-api")]
    public class PullRequestCommentsController : ControllerBase
    {
        private readonly IPullRequestCommentsRepository _repository;

        public PullRequestCommentsController(IPullRequestCommentsRepository repository)
        {
            _repository = repository;
        }

        [HttpGet("{owner}/{repository}/pulls/{pullRequestId}/comments")]
        public async Task<ActionResult<GetPullRequestCommentsResponseBody>> GetComments(string owner, string repository, string pullRequestId)
        {
            var comments = await _repository.FindAllBy(pullRequestId);
            return Ok(new GetPullRequestCommentsResponseBody {Comments = comments});
        }
    }
}

File: backend-kotlin/src/main/kotlin/pl/zycienakodach/eventmodeling/githubclone/slices/read/pullrequestcomments/presentation/PullRequestCommentsController.kt
package pl.zycienakodach.eventmodeling.githubclone.slices.read.pullrequestcomments.presentation

import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController
import pl.zycienakodach.eventmodeling.githubclone.shared.domain.RepositoryId
import pl.zycienakodach.eventmodeling.githubclone.slices.read.pullrequestcomments.application.PullRequestCommentsRepository

@ConditionalOnProperty(value = ["event-modeling.github-clone.slices.read.pull-request-comments.enabled"], havingValue = "true")
@RequestMapping("/rest-api")
@RestController
class PullRequestCommentsController(private val repository: PullRequestCommentsRepository) {

    @GetMapping("/{owner}/{repository}/pulls/{pullRequestId}/comments")
    fun getPullRequestComments(
        @PathVariable owner: String,
        @PathVariable repository: String,
        @PathVariable pullRequestId: String,
    ) = GetPullRequestCommentsResponseBody(this.repository.findAllBy(pullRequestId))

}
